src/test/java/org/opensearch/security/securityconf/FlattenedActionGroupsTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.securityconf;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ActionGroupsV7;
+
+public class FlattenedActionGroupsTest {
+    @Test
+    public void basicTest() throws Exception {
+        TestActionGroup testActionGroups = new TestActionGroup().with("Z", "C", "A")
```

Comment:
nit: IMO this could be more readable if these are labeled as group or action. i.e. groupA instead of A or actionA1 instead of Ai.

I think it would also be nice to define a test action group like new TestActionGroup().name("groupZ").allowedActions("groupC", "groupA");

Reply:
I have now made the `TestActionGroup` API more fluent. Additionally, I have added comments to the action group creation to indicate what elements get further expanded and what elements are leaves. 

However, I would like to keep the opaque naming to make clear that the resolver code does not put any semantics into the naming. In the end, it does not matter to the code what string is an action group and what is an action. It just resolves what it can resolve.